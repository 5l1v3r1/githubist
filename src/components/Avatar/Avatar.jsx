// @flow

import React from 'react';
import cx from 'classnames';

import styles from './Avatar.scss';

type Props = {
  url: string,
  alt: string,
  size?: 'default' | 'small',
};

const Avatar = ({ url, alt, size }: Props) => {
  const classNames = cx(styles.avatar, {
    [styles.default]: size === 'default',
    [styles.small]: size === 'small',
  });

  return <img src={url} alt={alt} className={classNames} />;
};

Avatar.defaultProps = {
  size: 'default',
};

export default Avatar;
