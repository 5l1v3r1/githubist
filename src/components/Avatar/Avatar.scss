@import '../../theme/theme';

.avatar {
  border-radius: 50%;
}

.default {
  width: 150px;
  height: 150px;
}

.small {
  width: 75px;
  height: 75px;
}
