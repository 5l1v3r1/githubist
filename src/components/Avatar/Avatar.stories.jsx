// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';

import Avatar from './Avatar';

storiesOf('Avatar', module)
  .add('default', () => <Avatar url="https://via.placeholder.com/150" alt="Default image" />)
  .add('small', () => (
    <Avatar url="https://via.placeholder.com/150" alt="Default image" size="small" />
  ));
