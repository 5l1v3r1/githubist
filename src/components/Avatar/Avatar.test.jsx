// @flow

import React from 'react';
import { shallow } from 'enzyme';

import Avatar from './Avatar';

it('renders correctly', () => {
  const wrapper = shallow(<Avatar url="https://via.placeholder.com/150" alt="Default image" />);

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly as small', () => {
  const wrapper = shallow(
    <Avatar url="https://via.placeholder.com/150" alt="Default image" size="small" />,
  );

  expect(wrapper).toMatchSnapshot();
});
