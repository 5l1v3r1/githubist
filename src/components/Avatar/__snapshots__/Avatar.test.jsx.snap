// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<img
  alt="Default image"
  className=""
  src="https://via.placeholder.com/150"
/>
`;

exports[`renders correctly as small 1`] = `
<img
  alt="Default image"
  className="undefined"
  src="https://via.placeholder.com/150"
/>
`;
